src/main/java/lotto/domain/Ball.java

```diff
@@ -1,24 +1,47 @@
 package lotto.domain;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
-public class Ball {
-    public static final int MINIMUM_NUMBER = 1;
-    public static final int MAXIMUM_NUMBER = 45;
+public class Ball implements Comparable<Ball> {
+    private static final int MINIMUM_NUMBER = 1;
+    private static final int MAXIMUM_NUMBER = 45;
     private static final String ERROR_LOTTO_NUMBER = "로또 숫자가 아닙니다";
+    public static final List<Ball> cacheBalls = new ArrayList<>();
 
     private final int number;
 
-    public Ball(final int number) {
+    static {
+        cacheBalls.addAll(IntStream.range(MINIMUM_NUMBER, MAXIMUM_NUMBER + 1)
+                .mapToObj(Ball::new)
+                .collect(Collectors.toList()));
+    }
```

Comment:
아래와 같이 사용하면 매직넘버를 줄일 수 있습니다.😊

```suggestion
    static {
        cacheBalls.addAll(IntStream.rangeClosed(MINIMUM_NUMBER, MAXIMUM_NUMBER )
                .mapToObj(Ball::new)
                .collect(Collectors.toList()));
    }
```